### PR TITLE
[Doc] Add docs for ProtobufNative Schema

### PR DESCRIPTION
**Pull request by Anonymitaet**
Add docs based on https://github.com/apache/pulsar/pull/8372 and update the green-highlighted parts.
schema doc architecture

### CODE REVIEW

**Comment by Anonymitaet** [comment]
@congbobo184 could u pls help review? Thank you.

**Review by congbobo184 — APPROVED** [review]
LGTM

File: site2/docs/schema-understand.md [hunk 1]
```diff
@@ -287,25 +289,36 @@ This example shows how to construct a key/value schema and then use it to produc
...
-This allows Pulsar:
+##### Type
 
-* to use same tools to manage schema definitions
+`struct` schema supports `AvroBaseStructSchema` and `ProtobufNativeSchema`.
 
-* to use different serialization/deserialization methods to handle data
+|Type|Description|
+---|---|
+`AvroBaseStructSchema`|Pulsar uses [Avro Specification](http://avro.apache.org/docs/current/spec.html) to declare the schema definition for `AvroBaseStructSchema`, which supports  `AvroSchema`, `JsonSchema`, and `ProtobufSchema`. <br><br>This allows Pulsar:<br>- to use same tools to manage schema definitions<br>- to use different serialization or deserialization methods to handle data|
```
Comment (Jennifer88huang) [inline site2/docs/schema-understand.md:304]: ... to use the same tools...

File: site2/docs/schema-understand.md [hunk 2]
```diff
@@ -287,25 +289,36 @@ This example shows how to construct a key/value schema and then use it to produc
...
+`ProtobufNativeSchema`|`ProtobufNativeSchema` is based on protobuf native Descriptor. <br><br>This allows Pulsar:<br>- to use native protobuf-v3 to serialize or deserialize data<br>- to use `AutoConsume` to deserialize data.
 
-There are two methods to use `struct` schema：
+##### Usage
+
+Pulsar provides three methods to use the `struct` schema:
```
Comment (Jennifer88huang) [inline site2/docs/schema-understand.md:309]: Technically, it's 100% correct.
I suggest that we use "Pulsar provides the following methods to ...."
When people add more methods below, they usually forget to update the specific number.

**Comment by Anonymitaet** [comment]
@Jennifer88huang thanks for your suggestions. I've incorporated your comments, PTAL.

**Review by Jennifer88huang — APPROVED** [review]

**Review by sijie — APPROVED** [review]

**Pull request merged**